pkg/bitwarden/cli_test.go

```diff
@@ -0,0 +1,561 @@
+package bitwarden
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestNewCliClientWithRun(t *testing.T) {
+	testCases := []struct {
+		name               string
+		username           string
+		password           string
+		run                func(cmd string, args ...string) ([]byte, error)
+		expectedSession    string
+		expectedSavedItems []Item
+		expectedError      error
+	}{
+		{
+			name:     "basic case",
+			username: "u",
+			password: "p",
+			run: func(cmd string, args ...string) ([]byte, error) {
+				if cmd == "bw" && reflect.DeepEqual(args, []string{"login", "u", "p", "--response"}) {
+					return []byte(`{
+  "success": true,
+  "data": {
+    "noColor": false,
+    "object": "message",
+    "title": "You are logged in!",
+    "message": "\nTo unlock your vault, set ...",
+    "raw": "not-going-to-tell-you=="
+  }
+}`), nil
+				} else if cmd == "bw" && reflect.DeepEqual(args, []string{"--session", "not-going-to-tell-you==", "list", "items"}) {
+					return []byte(`[
+  {
+    "object": "item",
+    "id": "id1",
+    "organizationId": "org1",
+    "folderId": null,
+    "type": 2,
+    "name": "unsplash.com",
+    "notes": null,
+    "favorite": false,
+    "fields": [
+      {
+        "name": "API Key",
+        "value": "value1",
+        "type": 0
+      }
+    ],
+    "secureNote": {
+      "type": 0
+    },
+    "collectionIds": [
+      "id1"
+    ],
+    "revisionDate": "2019-10-11T23:33:21.970Z"
+  },
+  {
+    "object": "item",
+    "id": "id2",
+    "organizationId": "org1",
+    "folderId": null,
+    "type": 2,
+    "name": "my-credentials",
+    "notes": "important notes",
+    "favorite": false,
+    "secureNote": {
+      "type": 0
+    },
+    "collectionIds": [
+      "id2"
+    ],
+    "attachments": [
+      {
+        "id": "a-id1",
+        "fileName": "secret.auto.vars",
+        "size": "161",
+        "sizeName": "161 Bytes",
+        "url": "https://cdn.bitwarden.net/attachments/111/222"
+      }
+    ],
+    "revisionDate": "2019-04-04T03:43:19.503Z"
+  }
+]`), nil
+				}
+				return nil, fmt.Errorf("unexpected call")
+			},
+			expectedSession: "not-going-to-tell-you==",
```

Comment:
This test is at the wrong level of abstraction. `NewClient()` can continue to work perfectly while the underlying cache implementation changes. Furthermore, if there is something _other_ than a CLI client, it may be implemented in a different way altogether. This should be a test for the `loginAndListItems()`

Reply:
Done.